base/inference.jl

```diff
@@ -191,7 +193,7 @@ mutable struct InferenceState
                     vararg_type = rewrap(vararg_type, linfo.specTypes)
                 end
                 s_types[1][la] = VarState(vararg_type, false)
-                src.slottypes[la] = widenconst(vararg_type)
+                src.slottypes[la] = vararg_type
```

Comment:
I think you can drop this part of the change. I don't foresee any need for inference or optimization to examine `slottypes` (it should be strictly suboptimal information relative to the dataflow-sensative types) – I think this is strictly consumed by codegen.